syntax = "proto3";

package job;

service Job {
    rpc SendJob (JobMetrics) returns (JobSpec) {}
}

message JobMetrics {
    string clientIP = 1;
    int32 cpuCount = 3;
    int32 memoryCount = 4;
}

message Provider {
    string providerID = 1;
    string providerIP = 2;
}

message JobSpec {
    Provider headProvider = 1;
    repeated Provider providers = 2;
}


message Empty {}

